File: Klanvision_backend_node/models/BlogPost.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const BlogPost = sequelize.define('BlogPost', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  title: {
    type: DataTypes.STRING
  },
  category: {
    type: DataTypes.STRING
  },
  author: {
    type: DataTypes.STRING
  },
  date: {
    type: DataTypes.STRING
  },
  readTime: {
    type: DataTypes.STRING,
    field: 'read_time'
  },
  views: {
    type: DataTypes.INTEGER,
    defaultValue: 0
  },
  image: {
    type: DataTypes.TEXT('long')
  },
  excerpt: {
    type: DataTypes.TEXT
  },
  content: {
    type: DataTypes.TEXT
  },
  status: {
    type: DataTypes.STRING
  },
  authorLink: {
    type: DataTypes.STRING,
    field: 'author_link'
  }
}, {
  tableName: 'blog_posts',
  timestamps: false
});

export default BlogPost;
export { BlogPost };

File: Klanvision_backend/src/main/java/com/klanvision/backend/model/BlogPost.java
package com.klanvision.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "blog_posts")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BlogPost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String excerpt;
    
    @Column(columnDefinition = "LONGTEXT")
    private String content;
    
    private String category;
    private String author;
    private String date;
    private String readTime;
    private String status;
    private String image;
    private int views;
    @Column(name = "author_link")
    private String authorLink;
}
